spec/rspec/core/formatters_spec.rb

```diff
@@ -137,6 +138,16 @@ module RSpec::Core::Formatters
           }.to change { loader.formatters.length }
         end
       end
+
+      context "When a custom formatter exists" do
+        before { loader.add Custom::AGeneralFormatter, output }
+
+        it "adds a subclass of that formatter for the same output target" do
+          expect {
+            loader.add Custom::AMoreSpecificFormatter, output
+          }.to change { loader.formatters.length }
```

Comment:
I've tried that, but it doesn't work. None of the formatters are added.

and if you do instead

```ruby
generic_formatter = BaseFormatter
specific_formatter = Class.new(generic_formatter)
```
Only the generic formatter is added (looks like classes don't get added unless they've registered).

The following works though (and doesn't require the custom formatter classes)

```ruby
specific_formatter = RSpec::Core::Formatters::JsonFormatter
generic_formatter = specific_formatter.superclass
```

I'll add those changes.